asv_bench/benchmarks/benchmarks.py

```diff
@@ -820,3 +832,106 @@ def time_index(self, shape):
 
 
 from .utils import setup  # noqa: E402, F401
+
+
+class NumericSeriesIndexing:
+
+    params = [
+        (np.int64, np.uint64, np.float64),
+        ("unique_monotonic_inc", "nonunique_monotonic_inc"),
+    ]
+    param_names = ["dtype", "index_structure"]
+
+    def setup(self, dtype, index_structure):
+        N = 10**6
+        indices = {
+            "unique_monotonic_inc": Index(range(N), dtype=dtype),
+            "nonunique_monotonic_inc": Index(
+                list(range(55)) + [54] + list(range(55, N - 1)), dtype=dtype
+            ),
+        }
+        self.data = Series(np.random.rand(N), index=indices[index_structure])
+        self.array = np.arange(10000)
+        self.array_list = self.array.tolist()
+
+    def time_getitem_scalar(self, index, index_structure):
+        self.data[800000]
+
+    def time_getitem_slice(self, index, index_structure):
+        self.data[:800000]
+
+    def time_getitem_list_like(self, index, index_structure):
+        self.data[[800000]]
+
+    def time_getitem_array(self, index, index_structure):
+        self.data[self.array]
+
+    def time_getitem_lists(self, index, index_structure):
+        self.data[self.array_list]
+
+    def time_iloc_array(self, index, index_structure):
+        self.data.iloc[self.array]
+
+    def time_iloc_list_like(self, index, index_structure):
```

Comment:
We already have loc/iloc benchmarks. Could you check for duplication in test cases.

Reply:
loc/iloc benchmarks currently present are for dataframe. The added benchmark is for Series objects, Would it make sense  to benchmark dataframe and series seperately?. 
Pandas uses 2 seperate benchmarking classes NumericSeriesIndexing and DataFrameNumericIndexing for series and dataframes.

Reply:
yes, it makes sense